packages/core/src/lib/code-snippet.svelte

```diff
@@ -0,0 +1,233 @@
+ */
+export let showCopyButton = true;
+
+/**
+ * Some prism languages have dependencies. For example, C++ requires C. If the
+ * passed `language` has dependencies, they must be included here to be loaded.
+ *
+ * See: https://prismjs.com/plugins/autoloader/
+ */
+export let dependencies: string[] = [];
+
+/** Additional CSS classes to pass to the aside. */
+let extraClasses: cx.Argument = '';
+export { extraClasses as cx };
+
+let element: HTMLElement | undefined;
+
+$: copyButtonLabel = 'Copy';
+$: copyButtonIcon = 'content-copy' as IconName;
+
+const dispatch = createEventDispatcher<{
+  copy: { succeeded: boolean; message: string };
+}>();
+
+const copyToClipboard = async () => {
+  try {
+    await navigator.clipboard.writeText(code);
+
+    copyButtonLabel = 'Copied';
+    copyButtonIcon = 'check';
+
+    dispatch('copy', {
+      succeeded: true,
+      message: 'Successfully copied snippet to the clipboard',
+    });
+  } catch {
+    copyButtonLabel = 'Failed';
+    copyButtonIcon = 'alert';
+
+    dispatch('copy', {
+      succeeded: false,
+      message: 'Failed to copy snippet to the clipboard',
+    });
+  }
+
+  window.setTimeout(() => {
+    copyButtonLabel = 'Copy';
+    copyButtonIcon = 'content-copy';
+  }, 2000);
+};
+
+const highlight = () => {
+  if (element) {
+    Prism.highlightElement(element);
+  }
+};
+
+const formatCode = (input: string): string => {
+  const htmlEntities: Record<string, string> = {
+    '<': '&lt;',
+    '>': '&gt',
+    '/': '&#47;',
+  };
+
+  let formattedCode = input;
+  for (const [key, value] of Object.entries(htmlEntities)) {
+    if (formattedCode) {
+      formattedCode = formattedCode.replaceAll(key, value);
+    }
+  }
+
+  return formattedCode;
+};
+
+$: {
+  if (element && code) {
+    element.innerHTML = formatCode(code);
+    highlight();
+  }
+}
+
+onMount(async () => {
+  try {
+    /**
+     * Load the themes, for vite to properly discover them we cannot use string templates,
+     * so we use a switch statement for our supported themes.
+     */
+    switch (theme) {
+      case 'vs': {
+        await import('prism-themes/themes/prism-vs.min.css');
+        break;
+      }
+      case 'vsc-dark-plus': {
+        await import('prism-themes/themes/prism-vsc-dark-plus.min.css');
+        break;
+      }
+      default: {
+        // Unsupported theme
+        break;
+      }
+    }
+
+    /**
+     * After the HTML is loaded in the DOM, we can use the autoloader to manage
+     * scanning for languages and including the components properly
+     */
+    await import(
+      // @ts-expect-error no type declaration for this JS file
+      'prismjs/plugins/autoloader/prism-autoloader'
+    );
+
+    // Make sure the autoloader knows where to find our languages
+    (Prism.plugins.autoloader as { languages_path: string }).languages_path =
+      // TODO: do we want a better way to get the version?
+      `https://cdnjs.cloudflare.com/ajax/libs/prism/${PRISM_VERSION}/components/`;
```

Comment:
Hate to say it, but highlight.js looks to have a much better ESM / dynamic [import story](https://highlightjs.org/#usage) that Prism, and looks to be more actively maintained. The third-party CDN approach still leaves a bad taste in my mouth

That being said, the best way to get the Prism version would be to read it from `package.json` in `vite.config.ts` and inject it into the build via the [define](https://vitejs.dev/config/shared-options.html#define) option

Reply:
Been playing with highlight.js for a minute, and it's not exactly going well 😢

Reply:
I think @mrloureed had a similar issue with it.

Reply:
Also I tried using define with vite but it looks like vite wasn’t properly replacing it in the build, which meant I had to re-declare it in storybook which we definitely don’t want users needing to do. Glad to re-examine that, but is that better than just grabbing the library version from the library itself? The nice thing there is I didn’t have to do any string replacements.